### PR TITLE
issue 271 true | IContainChainsOptions 

### PR DESCRIPTION
**Pull request by rienafairefr**
passing an object implementing IContainChainsOptions instead of a boolean seems to solve a small type error

### CODE REVIEW

**Comment by danmarshall** [comment]
Thanks for the PR! BTW - Which version of TypeScript caught this?

**Comment by danmarshall** [comment]
or was it a runtime error?

**Comment by rienafairefr** [comment]
It was a compile error, caught by typescript 2.4 AFAIK,

well, To be frank, I just started typescript a few days ago. I think it was on typescript 2.4.x , failed with this error, I _think_ I downgraded to 2.1.6 and then it compiled ^^
Right now, clean build from my fork and this repo master branch, upgraded nodejs to 8.1, new folder, typescript installed seems to be 2.4.1 and the only build failure seems to be this one. Still trying to get the hang of this (I'm more of a python or C# expert)

File: src/core/chain.ts [hunk 1]
```diff
@@ -294,7 +294,7 @@
             chainsByLayer[layer].sort((a: IChain, b: IChain) => { return b.pathLength - a.pathLength });
 
             if (opts.contain) {
-                var containedChains = getContainment(chainsByLayer[layer], opts.contain);
+                var containedChains = getContainment(chainsByLayer[layer], {alernateWindings:true});
```
Comment (danmarshall) [inline src/core/chain.ts]: This `{alternateWindings:true}` wouldn't allow someone to pass in their own options in `opts.contains`, for example, if they passed in `false`.
Also there's a typo: `alernate...`  ;-)

Comment (danmarshall) [inline src/core/chain.ts]: The mixed type `boolean | IContainChainsOptions` was for backwards compat. But I see that `getContainment` wants only the object. Maybe we add a local var to make sure that we cast only the object as value? And I suppose the default would be `false`.

Nice catch BTW

Comment (danmarshall) [inline src/core/chain.ts]: will this satisfy it?
```
var containChainsOptions: IContainChainsOptions = isObject(opts.contain) ? opts.contain as IContainChainsOptions : { alernateWindings: false };
var containedChains = getContainment(chainsByLayer[layer], containChainsOptions);
```

Comment (rienafairefr) [inline src/core/chain.ts]: no merit, the compiler did the work :-)

I think that might work. alernatewinding is indeed a typo in the code, we keep it for backward compatibility ?

File: src/core/chain.ts [hunk 2]
```diff
@@ -315,7 +316,7 @@
     /**
      * @private
      */
-    function getContainment(allChains: IChain[], opts: IContainChainsOptions) {
+    function getContainment(allChains: IChain[], opts: boolean | IContainChainsOptions) {
```
Comment (danmarshall) [inline src/core/chain.ts]: here we don't want the boolean.

**Pull request merged**